### ISSUE TITLE
Enumerate all distinct values

### ISSUE THREAD

**Issue by Synthetica9** [pa-ba/equivalence#7]
As the title states. I have a function as follows:

```haskell
mergeEquivalent :: (Ord a) => (a -> a -> Bool) -> (a -> a -> a) -> ([a] -> [a])
mergeEquivalent (===) (+) xs = let
  -- https://github.com/quchen/articles/blob/master/2018-11-22_zipWith_const.md
  indices = zipWith const [0..] xs -- Because just [0..len] generates [0] when xs == []!
  combis = pairs indices

  update :: IntSet.IntSet -> (Int, Int) -> EQ.EquivM _ _ Int IntSet.IntSet
  update = \s (i1, i2) -> do
    eq <- EQ.equivalent i1 i2
    if eq
      then pure s
      else do
        x1 <- EQ.classDesc i1
        x2 <- EQ.classDesc i2
        if (x1 === x2)
          then do
            EQ.equate i1 i2
            pure $ IntSet.delete i1 s
          else
            pure s

  in EQ.runEquivM (xs !!) (+) $ do
    s <- foldM update (IntSet.fromList indices) combis
    classes <- for (IntSet.toList s) EQ.classDesc
    pure classes
```

But I have to manually keep track of the classes that exists. Is there a possibility that a function would be added that does this automatically? If I understand the implementation correctly, this should be possible in _O(n)_

An alternative would be to implement something like [`redundant :: v -> m Bool`](https://hackage.haskell.org/package/union-find-0.2/docs/Data-UnionFind-ST.html#v:redundant) so that I can manually filter.

TL;DR:

I want an `allElements :: m [v]`

**Comment by andreasabel** [comment]
Fixed by 
- #13 

Scheduled for 0.4.1.
